### DISCUSSION TITLE
Ruby-build 2.7.3 not able to download from internet

### DISCUSSION THREAD

**Discussion by lkkaur-os** [rbenv/ruby-build#1982] [Build failures]
I am mirroring ruby-build to artifactory with the aim to avoid taking packages from internet but I am unable to do it. Is there any way to do it?

I am using command to build in dockerfile - RUN ruby-build 2.7.3 /usr/local/bin  but it gives below error

```
Downloading ruby-2.7.3.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.3.tar.bz2
error: failed to download ruby-2.7.3.tar.bz2
```

Is there any way to build this 2.7.3 version from the mirrored artifactory instead of downloading it from the internet? 

#753 #803

**Comment by mislav** [comment]
ruby-build will [use its own mirror by defaul](https://github.com/rbenv/ruby-build#package-mirrors)t. The fact that downloads don't work in your container points to a possible networking problem in your container, not caused by ruby-build.

You can also set `RUBY_BUILD_MIRROR_URL` to specify a mirror that might be local to your network, but then you have to maintain your mirror. I would suggest solving your network issues as an easier solution.

**Reply by lkkaur-os** [reply]
I tried setting RUBY_BUILD_MIRROR_URL in  dockerfile also, but then it throws error saying that checksum mismatch: ruby-2.7.3.tar.bz2(file is corrupt)

Does anything else needs to be set with  RUBY_BUILD_MIRROR_URL variable?

**Reply by mislav** [reply]
No, nothing else needs to be set. Ruby-build will construct the full URL of the download like this `$RUBY_BUILD_MIRROR_URL/<SHA2>`, where `<SHA2>` is the SHA-2 checksum of the file to download. The URL must point to a valid file. According to your message, no valid file was found at the mirror location. I'm sorry, but I can't help much here.
